internal/k8s/gatewayclient/gatewayclient.go

```diff
@@ -474,3 +477,14 @@ func (g *gatewayClient) HasManagedDeployment(ctx context.Context, gw *gateway.Ga
 	}
 	return false, nil
 }
+
+func (g *gatewayClient) GetReferencePoliciesInNamespace(ctx context.Context, namespace string) ([]gateway.ReferencePolicy, error) {
+	list := &gateway.ReferencePolicyList{}
+	if err := g.Client.List(ctx, list, client.InNamespace(namespace)); err != nil {
+		if k8serrors.IsNotFound(err) {
```

Comment:
nit: we shouldn't need this conditional branch since a `List` invocation is just going to return a list with no items and shouldn't ever hand back `IsNotFound` -- so think we just need the `err != nil` check.